doc/nomenclature.md

```diff
@@ -2,9 +2,61 @@
 
 ## Table of Contents
 
+- [A](#a)
+    - [AST][ast]
+- [M](#m)
+    - [Mutagenesis][mutagenesis]
+    - [Mutant][mutant]
+    - [Mutation][mutation]
+    - [Mutator][mutator]
+- [S](#s)
+    - [Subject][subject]
 - [T](#t)
-  - [Tracer][tracer]
-  - [Trace][trace]
+    - [Tracer][tracer]
+    - [Trace][trace]
+
+
+## A
+
+### AST
+
+Acronym for [Abstract Syntax Tree][ast-definition]. It is a tree representation of the abstract
+syntactic structure of code. It is what Infection parses the code into in order to operate on it.
+
+
+## M
+
+### Mutagenesis
+
+Process of creating a mutant from the original program.
+
+
+### Mutant
+
+New program that differs from the original by applying a mutation.
+
+
+### Mutation
+
+The result of applying a mutator to a representation of a subject (e.g. AST, bytecode) and represents a change to be applied.
+
+In the Mutation Testing literature, the terms "mutant" and "mutation" are often interchanged as in most cases
+one wants to refer to the change which both terms capture well.
+
+
+### Mutator
```

Comment:
When I hear about `Mutator` I always think first about a function that changes the internal state. I wish we called these something other than mutator, like  mutagen.